src/ContainerDetails.jsx

```diff
@@ -12,6 +12,18 @@ const render_container_state = (container) => {
     return cockpit.format(_("Exited"));
 };
 
+const render_container_published_ports = ({ ports }) => {
+    let result = [];
+    if (!ports)
+        return result;
+    for (let i = 0; i < ports.length; ++i)
+        result.push(
+            <React.Fragment key={ ports[i].protocol + ports[i].host_port + ports[i].container_port }>
```

Comment:
So you said you want to use the index here as well, right? I suppose this could use the intrinsic values if they are set, and the index if any of them are empty?

Reply:
In details the index is not necessary, because here the values are already set for sure, as these are running containers.

Reply:
Summary from IRC: Let's keep a `this.id_counter`, increase it when creating a new ports input line, and use that as an id/key for the `{containerPort, hostPort, protocol, id}` object. This will be an invariant intrinsic property that is suitable as key, and survives reordering from deletion.